### PR TITLE
Prevent crash in case there are no contracts

### PR DESCRIPTION
**Pull request by mestorlx**
This change fix issue #5007 
however I am not sure about the gas approach.
@chriseth can you please comment.

Will add test and update changelog later

### CODE REVIEW

File: solc/CommandLineInterface.cpp [hunk 1]
```diff
@@ -994,16 +994,22 @@ void CommandLineInterface::handleAst(string const& _argStr)
...
-			auto ret = GasEstimator::breakToStatementLevel(
-				GasEstimator(m_evmVersion).structuralEstimation(*m_compiler->runtimeAssemblyItems(m_compiler->lastContractName()), asts),
-				asts
-			);
-			gasCosts = ret;
+		for (auto const& contract : m_compiler->contractNames())
+			{
+			if (m_compiler->runtimeAssemblyItems(contract))
+			{
+				//NOTE: keep the local variable `ret` to prevent a Heisenbug that could happen on certain mac os platform.
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]: Can be deleted now.

File: solc/CommandLineInterface.cpp [hunk 2]
```diff
@@ -994,16 +994,22 @@ void CommandLineInterface::handleAst(string const& _argStr)
...
+				//See: https://github.com/ethereum/solidity/issues/3718 for details.
+				auto ret = GasEstimator::breakToStatementLevel(
+					GasEstimator(m_evmVersion).structuralEstimation(*m_compiler->runtimeAssemblyItems(contract), asts),
+					asts
+				);
+				if (gasCosts.empty())
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]: Why not just
```
for (auto const& it: ret)
  gasCosts[it.first] += it.second;
```

File: solc/CommandLineInterface.cpp [hunk 3]
```diff
@@ -994,16 +994,22 @@ void CommandLineInterface::handleAst(string const& _argStr)
...
-		{
-			//NOTE: keep the local variable `ret` to prevent a Heisenbug that could happen on certain mac os platform.
-			//See: https://github.com/ethereum/solidity/issues/3718 for details.
-			auto ret = GasEstimator::breakToStatementLevel(
-				GasEstimator(m_evmVersion).structuralEstimation(*m_compiler->runtimeAssemblyItems(m_compiler->lastContractName()), asts),
-				asts
-			);
-			gasCosts = ret;
+		for (auto const& contract : m_compiler->contractNames())
+			{
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]: No braces needed.

File: solc/CommandLineInterface.cpp (hunk 1 above)
Comment (mestorlx) [reply solc/CommandLineInterface.cpp]: done

**Comment by mestorlx** [comment]
Comments addressed.
@chriseth Please review.

File: Changelog.md [hunk 4]
```diff
@@ -132,6 +132,7 @@ Bugfixes:
  * Type Checker: Fix internal error when array index is not an unsigned.
  * Type System: Allow arbitrary exponents for literals with a mantissa of zero.
  * Parser: Fix incorrect source location for nameless parameters.
+ * Command Line Interface: Fix internal error when compiling stdin with no content
```
Comment (chriseth) [inline Changelog.md]: The important aspect is that this concerns the ast output

File: solc/CommandLineInterface.cpp (hunk 1 above)
Comment (chriseth) [reply solc/CommandLineInterface.cpp]: the comment is still there

**Review by chriseth — APPROVED** [review]

**Pull request merged**